src/devices/Dhtxx/DhtSensor.cs

```diff
@@ -46,48 +47,25 @@ private void Wait(double milliseconds)
         public bool IsLastReadSuccessful { get; internal set; }
 
         /// <summary>
-        /// Get the last read temperature in Celsius
+        /// Get the last read temperature
         /// </summary>
         /// <remarks>
         /// If last read was not successfull, it returns double.MaxValue
         /// </remarks>
-        public double Temperature => (_dhtType == DhtType.Dht11) ? GetTempDht11() : GetTempDht22();
+        public Temperature Temperature => Temperature.FromCelsius((_dhtType == DhtType.Dht11) ? GetTempDht11() : GetTempDht22());
```

Comment:
I think it's better to return double, because I don't need Fahrenheit and Kelvin every time when I read the temperature.

Reply:
Although it doesn't take much time for Raspberry Pi to do multi-step operations, data conversion is only a few cases. As far as I know, there are only five countries that use Fahrenheit. The largest country is the United States. Most countries in the world use Celsius.

Reply:
I still think we should use the extension method. I reserve my opinion and wait for others to discuss it. 😄

Reply:
The temperature is internally stored as Celsius as that is the most common unit.

Reply:
@ZhangGaoxing we will consider adding implicit conversion to float in the future but we didn't want to make this decision just yet without a strong case (would be hard to remove in the future if we have reasons not to do so). I'm personally prefer interfaces and more functional approach but there are some hidden costs with using interfaces and also it is hard to make a case of them being useful (especially when they have only one property which is a double)